addons/binding/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/ThirdGeneration/ThirdGenerationHandler.java

```diff
@@ -0,0 +1,767 @@
+            updateChannelValues();
+        }, 10, config.refreshInternalInSeconds, TimeUnit.SECONDS);
+    }
+
+    /**
+     * The API supports the resolution of multiple values at a time
+     *
+     * Therefore this methods builds one request to gather all information for the current inverter.
+     * The list contains all channels as defined in {@link ThirdGenerationMappingInverterToChannel} for the
+     * current inverter
+     *
+     */
+    void updateChannelValues() {
+        Map<String, List<ThirdGenerationChannelMappingToWebApi>> channelList = ThirdGenerationMappingInverterToChannel
+                .getModuleToChannelsMappingForInverter(inverterType);
+        JsonArray updateMessageJsonArray = new JsonArray();
+        /*
+         * Build the message to send to the inverter
+         */
+        for (String moduleId : channelList.keySet()) {
+            JsonObject moduleJsonObject = new JsonObject();
+            moduleJsonObject.addProperty("moduleid", moduleId);
+
+            JsonArray processdataNames = new JsonArray();
+            for (ThirdGenerationChannelMappingToWebApi processdata : channelList.get(moduleId)) {
+                processdataNames.add(processdata.processdataId);
+            }
+            moduleJsonObject.add("processdataids", processdataNames);
+            updateMessageJsonArray.add(moduleJsonObject);
+        }
+        /*
+         * Send the API request to get values for all channels
+         * Note: If one does not exist, all fail
+         */
+        ContentResponse updateMessageContentResponse;
+        try {
+            updateMessageContentResponse = executeHttpPost(PROCESSDATA, updateMessageJsonArray, sessionId);
+            if (updateMessageContentResponse.getStatus() == 404) {
+                /*
+                 * Module not found
+                 */
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        COMMUNICATION_ERROR_INCOMPATIBLE_DEVICE);
+                return;
+            }
+            if (updateMessageContentResponse.getStatus() == 503) {
+                /*
+                 * Communication error (e.g. during initial boot of the SCB)
+                 */
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        COMMUNICATION_ERROR_HTTP);
+                return;
+            }
+            if (updateMessageContentResponse.getStatus() == 401) {
+                /*
+                 * session not valid (timed out?)
+                 */
+                logger.info("Session expired");
+                try {
+                    authenticate();
+                } catch (Exception e) {
+                }
+                /*
+                 * Retry
+                 */
+                updateMessageContentResponse = executeHttpPost(PROCESSDATA, updateMessageJsonArray, sessionId);
+            }
+        } catch (Exception e) {
```

Comment:
Don't catch `Exception`, But if any catch `RuntimeException` and/or any specific exception if possible.

This comment applies to all cases were you catch `Exception`.